stdlib/public/Cxx/CxxSet.swift

```diff
@@ -43,3 +46,25 @@ extension CxxSet {
     return count(element) > 0
   }
 }
+
+public protocol CxxUniqueSet<Element>: CxxSet {
+  override associatedtype Element
+  override associatedtype Size: BinaryInteger
+  associatedtype RawMutableIterator: UnsafeCxxInputIterator
+    where RawMutableIterator.Pointee == Element
+  override associatedtype InsertionResult
+    where InsertionResult: CxxPair<RawMutableIterator, Bool>
+}
+
+extension CxxUniqueSet {
+  @inlinable
+  @discardableResult
+  public mutating func insert(
+    _ element: Element
+  ) -> (inserted: Bool, memberAfterInsert: Element) {
+    let insertionResult = self.__insertUnsafe(element)
```

Comment:
Worried this could be destroyed before L68 when it's accessed.

Reply:
`insertionResult` is a `std::pair<iterator, bool>`. The iterator itself won't be destroyed since we store it in `let rawIterator`. I think it is safe if `insertionResult` is destroyed before L68?